marconi-sidechain/doc/API.adoc

```diff
@@ -0,0 +1,890 @@
+          "properties": {
+            "blockNo": {
+              "type": "integer",
+              "minimum": 0
+            },
+            "blockTimestamp": {
+              "type": "string",
+              "minimum": 0,
+              "description": "timestamp in seconds"
+            },
+            "blockHeaderHash": {
+              "type": "string",
+              "pattern": "^[0-9a-f]{64}$"
+            },
+            "slotNo": {
+              "type": "integer",
+              "minimum": 0
+            },
+            "epochNo": {
+              "type": "integer",
+              "minimum": 0
+            }
+          },
+          "required": []
+        }
+      },
+      "required": [
+        "id",
+        "jsonrpc",
+        "result"
+      ]
+    },
+    {
+      "type": "object",
+      "properties": {
+        "id": {
+          "type": "integer"
+        },
+        "jsonrpc": {
+          "type": "string"
+        },
+        "error": {
+          "type": "object",
+          "properties": {
+            "code": {
+              "type": "integer"
+            },
+            "message": {
+              "type": "integer"
+            },
+            "data": {
+              "type": "integer"
+            }
+          },
+          "required": [
+            "code",
+            "message"
+          ]
+        }
+      },
+      "required": [
+        "id",
+        "jsonrpc",
+        "error"
+      ]
+    }
+  ]
+}
+```
+====
+
+==== Pre-conditions
+
+[NOTE]
+====
+In here, we'll document the conditions that need to met in order to get a successful response.
+====
+
+==== Post-conditions
+
+[NOTE]
+====
+In here, we'll document the properties of the successful response.
+====
+
+==== Method-specific errors
+
+This method should never return an error message.
+
+[NOTE]
+====
+In here, we'll document each possible method-specific error code, along with an explaining on when such an error occurs.
+====
+
+==== Rationale
+
+[NOTE]
+====
+In here, we'll document the rationale for having this method => why does the sidechain application need it.
+====
+
+
+=== getUtxosFromAddress
+
+Retrieves UTXOs of a given address until a given point in time (measured in slots).
+
+==== JSON Schema
+
+.HTTP Request body
+[%collapsible]
+====
+```JSON
+{
+  "$schema": "http://json-schema.org/draft-04/schema#",
+  "type": "object",
+  "properties": {
+    "jsonrpc": {
+      "type": "string"
+    },
+    "method": {
+      "type": "string"
+    },
+    "params": {
+      "type": "object",
+      "properties": {
+        "address": {
+          "type": "string",
+          "description": "Show UTxOs belonging to this Shelley address in bech32 format."
+        },
+        "createdAfterSlotNo": {
+          "type": "integer",
+          "minimum": 0,
+          "description": "Filter out UTxO that were created during or before that slot."
+        },
+        "unspentBeforeSlotNo": {
+          "type": "integer",
+          "minimum": 0,
```

Comment:
minimum should be 1, since this is a mandatory element

Reply:
`minimum` is not related to whether an element is mandatory or not. That's what the `required` field below is about.
However, `"minimum": 1` might make sense if `Slot 0` does not exist. However, I _think_ slot 0 exists afaik. Need to check.

Reply:
I found slot 0, confusing in some of our tests, as I am not sure if  there is such slot number.
And am looking forward to your answer, thnx